feat(stepper): initial design guidance (#26)
vmware-clarity/clarity-guidance: 1030-stepper-design-guidance.md

---
category: component design guidance
state: draft
created: 2024-10-16
updated: 2024-10-22
---

> # Stepper Component

Reviewer: ```suggestion
# Stepper Design Guidance
```

> The [stepper](https://clarity.design/documentation/stepper) is an accordion-like component that structures multi-step processes into two or more expanding panels that break up complex workflows.

Reviewer 2: ```suggestion
The [stepper](https://clarity.design/documentation/stepper) is an accordion-like component that organizes multi-step processes into two or more expanding panels, helping to simplify complex workflows.
```

> The [Stepper](https://clarity.design/documentation/stepper) is an extension of the base accordion component. It structures multi-step processes into two or more expanding panels that break up complex workflows.

Reviewer: ```suggestion
The [stepper](https://clarity.design/documentation/stepper) is an accordion-like component that structures multi-step processes into two or more expanding panels that break up complex workflows.
```

The fact that the step is an "extension" of the accordion is an implementation detail that will hopefully not be true in the future.

## Guidance

### Stepper Usage

> - **Must** use a stepper when users need to see the forms in-line with the rest of the page content.

Reviewer 2: ```suggestion
- Design teams **must** use a stepper when users need to see the forms in line with the rest of the page content, as this approach helps maintain context and flow, reducing cognitive load during multi-step processes.
```

> - **Must** use a stepper when you have complex and long work processes that may not fit in one viewport.

Reviewer: ```suggestion
- **May** use a stepper when you have complex and long work processes that may not fit in one viewport.
```

Other options exist such as the wizard.

> - **Must** use a stepper when users need more detail than step titles to understand the process.

Reviewer 2: ```suggestion
- Design teams **must** use a stepper when users require more detail than step titles to understand the process, as this provides necessary context and guidance for completing each step effectively.
```

> - **Must** use a stepper when you have many steps with small data set in each step. For longer forms with a fewer number of steps, use a [timeline](https://clarity.design/documentation/timeline) component.

Reviewer 2: ```suggestion
- Design teams **must** use a stepper when there are many steps with a small dataset in each step. For longer forms with fewer steps, use a [timeline](https://clarity.design/documentation/timeline) component to ensure clarity and organization in the workflow.
```

> - **Must** use a stepper when you have many steps with small data set in each step. For longer forms with a fewer number of steps use a [Timeline](https://clarity.design/documentation/timeline) component.

Reviewer: ```suggestion
- **Must** use a stepper when you have many steps with small data set in each step. For longer forms with a fewer number of steps, use a [timeline](https://clarity.design/documentation/timeline) component.
```

> - **Must** use a stepper when you need the capability to navigate freely to other parts of an application during the workflow. To block access to other parts of the application while completing the workflow, use a [wizard](https://clarity.design/documentation/wizard) component.

Reviewer 2: ```suggestion
- Design teams **must** use a stepper when users need the ability to navigate freely to other parts of the application during the workflow. If access to other parts of the application should be restricted while completing the workflow, use a [wizard](https://clarity.design/documentation/wizard) component instead.
```

> - **Should** use a stepper when you need a preview or summary of each step.

Reviewer 2: ```suggestion
- Design teams **should** use a stepper when a preview or summary of each step is needed, as this provides users with context and clarity about what to expect throughout the process.
```

> - **Must** use a stepper when you need the capability to navigate freely to other parts of an application during the workflow. To block access to other parts of the application while completing the workflow use a [Wizard](https://clarity.design/documentation/wizard) component.

Reviewer: ```suggestion
- **Must** use a stepper when you need the capability to navigate freely to other parts of an application during the workflow. To block access to other parts of the application while completing the workflow, use a [wizard](https://clarity.design/documentation/wizard) component.
```

> - **Should** use a stepper when you have more than two steps.

Reviewer 2: ```suggestion
- Design teams **should** use a stepper when there are more than two steps, as this helps users navigate through the process more effectively and understand their progress.
```

> - **May** use a stepper when you have complex and long work processes that might not fit in one viewport.

Reviewer 2: ```suggestion
- Design teams **may** use a stepper for complex and lengthy workflows that might not fit within one viewport, allowing users to manage the process in smaller, more digestible sections.
```

### Stepper Anatomy

> - **Must not** have too long content in each step.

Reviewer 2: ```suggestion
- Design teams **must not** include overly long content in each step, as this can overwhelm users and make it difficult to focus on the task at hand. Each step should present information clearly and concisely.
```

> - **Must** NOT have too long content in each step.

Reviewer: ```suggestion
- **Must not** have long content in each step.
```

> - Stepper panel header **should** contain a title and a description for the step.

Reviewer 2: ```suggestion
- The stepper panel header **should** contain a title and a description for the step, providing users with clear context and guidance on what is required at that stage of the process.
```
